amcl/src/amcl_node.cpp

```diff
@@ -393,71 +446,91 @@ AmclNode::AmclNode() :
+  global_loc_srv_ = node->create_service<std_srvs::srv::Empty>("global_localization",
+					 std::bind(&AmclNode::globalLocalizationCallback, this, std::placeholders::_1, std::placeholders::_2));
+  nomotion_update_srv_= node->create_service<std_srvs::srv::Empty>("request_nomotion_update",
+					 std::bind(&AmclNode::nomotionUpdateCallback, this, std::placeholders::_1, std::placeholders::_2));
+  set_map_srv_= node->create_service<nav_msgs::srv::SetMap>("set_map",
+					 std::bind(&AmclNode::setMapCallback, this, std::placeholders::_1, std::placeholders::_2));
+
+  // laser_scan_sub_ = new message_filters::Subscriber<sensor_msgs::msg::LaserScan>(nh_, scan_topic_, 100);
+  qos = rmw_qos_profile_default;
+  qos.depth = 100;
+  laser_scan_sub_ = node->create_subscription<sensor_msgs::msg::LaserScan>(scan_topic_, std::bind(&AmclNode::laserReceived, this, std::placeholders::_1), qos);
+  /*
+  laser_scan_filter_ =
+          new tf::MessageFilter<sensor_msgs::msg::LaserScan>(*laser_scan_sub_,
+                                                        *tf_,
+                                                        odom_frame_id_,
                                                         100);
-  laser_scan_filter_->registerCallback(boost::bind(&AmclNode::laserReceived,
+  laser_scan_filter_->registerCallback(std::bind(&AmclNode::laserReceived,
                                                    this, _1));
-  initial_pose_sub_ = nh_.subscribe("initialpose", 2, &AmclNode::initialPoseReceived, this);
+ */
+  qos.depth = 2;
+  initial_pose_sub_ = node->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>("initialpose", std::bind(&AmclNode::initialPoseReceived, this, std::placeholders::_1), qos);
 
   if(use_map_topic_) {
-    map_sub_ = nh_.subscribe("map", 1, &AmclNode::mapReceived, this);
-    ROS_INFO("Subscribed to map topic.");
+    map_sub_ = node->create_subscription<nav_msgs::msg::OccupancyGrid>("map", std::bind(&AmclNode::mapReceived, this, std::placeholders::_1), qos);
   } else {
     requestMap();
   }
   m_force_update = false;
 
+  /*
   dsrv_ = new dynamic_reconfigure::Server<amcl::AMCLConfig>(ros::NodeHandle("~"));
-  dynamic_reconfigure::Server<amcl::AMCLConfig>::CallbackType cb = boost::bind(&AmclNode::reconfigureCB, this, _1, _2);
+  dynamic_reconfigure::Server<amcl::AMCLConfig>::CallbackType cb = std::bind(&AmclNode::reconfigureCB, this, _1, _2);
   dsrv_->setCallback(cb);
+  */
 
   // 15s timer to warn on lack of receipt of laser scans, #5209
-  laser_check_interval_ = ros::Duration(15.0);
-  check_laser_timer_ = nh_.createTimer(laser_check_interval_, 
-                                       boost::bind(&AmclNode::checkLaserReceived, this, _1));
+  laser_check_interval_ = std::chrono::seconds(15);
+  check_laser_timer_ = node->create_wall_timer(laser_check_interval_,
+                                       std::bind(&AmclNode::checkLaserReceived, this));
```

Comment:
NIt: this should either be aligned with the line above or use 2-whitespace indent